bots/pr/src/main/java/org/openjdk/skara/bots/pr/CheckRun.java

```diff
@@ -443,6 +443,28 @@ private boolean updateClean(Commit commit) {
         return isClean || isCleanLabelManuallyAdded;
     }
 
+    private void updateMergeClean(Commit commit) {
+        boolean isClean = !commit.isMerge();
+        if (!isClean) {
+            isClean = true;
+            var commitMessageBody = localRepo.commitMessageBody(commit.hash());
+            if (commitMessageBody.isPresent()) {
+                var lines = commitMessageBody.get();
+                for (int i = 0; i < lines.size() - 1; i++) {
+                    if (lines.get(i).startsWith("diff") && lines.get(i + 1).startsWith("index")) {
+                        isClean = false;
+                        break;
+                    }
+                }
+            }
```

Comment:
I think this is doing the right thing, but the abstraction isn't right. If we were to ever implement this for HG, then we can't know what the output would look like here, and we can't assume it will be the same as for Git. The knowledge about how to interpret the output of Git needs to be encapsulated in `GitRepository`. Most of this method should move into a new method in `Repository`/`GitRepository`.

Thinking some more about this, don't we have enough information in the `Commit` instance to figure this out? If the commit doesn't have any patches, then we should be good to go right? If so, we don't need any new method on `Repository`.